mflowgen/components/subgraph.py

```diff
@@ -0,0 +1,552 @@
+#=========================================================================
+# subgraph.py
+#=========================================================================
+# Author : Alex Carsello
+# Date   : July 18, 2023
+#
+
+import copy
+import importlib
+import os
+import sys
+import yaml
+
+from mflowgen.utils import get_top_dir, read_yaml, write_yaml
+from mflowgen.components.step import Step
+from mflowgen.core.run import RunHandler as rh
+
+class Subgraph(Step):
```

Comment:
Should we just be using multiple inheritance from Step and Graph?

Reply:
There's a lot of repeated code here, right?

Reply:
Yes, I should remove most/all of it.